### ISSUE TITLE
Plans Next: Introduce snapshot tests for plans-grid-next components

### ISSUE THREAD

**Issue by chriskmnds** [Automattic/wp-calypso#82867]
### What

Snapshot tests for UI components in `plans-grid` (package).

### Why

Most UI components in `plans-grid` have more or less stabilised in terms of design/functionality. We can introduce some snapshot tests to guard from side effect breaking things. 

This was suggested by @aneeshd16 in Slack ( p1696919568941969-slack-CV2TX2PAN ) after reviewing https://github.com/Automattic/wp-calypso/pull/82715 - and rightly so, testing similar changes would be easier with snapshot tests (we'd immediately know of unexpected effects).

**Comment by chriskmnds** [comment]
@southp I wasn't sure under which Milestone in PT ( pau2Xa-59h-p2 ) this would fit best. It does relate to the package and its code hygiene, and for the most part, Milestone 0 has been the generic basis/backbone for most things. I can add it there, or else we create another? or keep this out of NPM refactor PT? 🤷‍♂️

**Comment by southp** [comment]
Good question. I think it fits the Plans to NPM project the best since implementing test suites like that often lead to some related dependency cleanning-up. In the meantime, I also don't think further bloating milestone #0 would be a good idea. Thus, Milestone 2 would be a good fit from my perspective. What do you think? :)

**Comment by chriskmnds** [comment]
> Thus, Milestone 2 would be a good fit from my perspective. What do you think? :)

Hmm I'm unsure about Milestone 2 @southp. It feels more like a Milestone 0 technically, or something that can come towards the end. Somehow Milestone 0 is the one about fixing the foundations, which we are iterating on while working on the rest. I think once we clean and split the props and refactor the components to be in functional form, then we'd have a good basis to introduce some snapshot tests. Maybe @jdc91 has more thoughts, as he's working on both of these? 🤔

**Comment by ddc22** [comment]
Maybe we can add a crude first version in milestone 0 (Basic logged out plans page snapshot without thinking too much with some generic props) of the snapshot since the prop cleanup is a bit tricky and can result in unintended changes. Then from there onwards we can progressively grow the snapshot with each milestone to cover more scenarios. 🤔 And we will have a record of the changes will be version controlled as we progress through the refactor.

**Comment by southp** [comment]
Yeah, it makes sense to me now that it belongs to milestone #0 and is closely related to the prop auditing work. 
@jdc91 if I understand you correctly, what you suggested is to implement one based on what we have now, so it could actually serve as the safenet alongside the prop refactoring work. That echoes deeply with my TDD itch, so I'd be excited to see it happen :) Would you have the bandwidth to accomplish it along side your current work?

**Comment by chriskmnds** [comment]
We can add this under Milestone 3 now (package extraction). This is the right time and place to consider these types of tests.
